src/librustc_typeck/check/op.rs

```diff
@@ -187,33 +187,23 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                             op.node.as_str(),
                             lhs_ty);
 
-                        match lhs_ty.sty {
-                            TypeVariants::TyRef(_, ref ty_mut) => {
-                                match ty_mut.ty.sty {
-                                    TypeVariants::TyRef(_, ref ty_mut) => {
-                                        match self.lookup_op_method(expr, ty_mut.ty, vec![rhs_ty_var],
-                                                    token::intern(name), trait_def_id,
-                                                    lhs_expr) {
-                                            Ok(_) => {
-                                                err.span_note(
-                                                    lhs_expr.span,
-                                                    &format!(
-                                                        "this is a reference of a reference to a \
-                                                        type that `{}` can be applied to, you need \
-                                                        to dereference this variable once for this \
-                                                        operation to work",
-                                                    op.node.as_str()));
-                                            },
-                                            _ => (),
-                                        }
-                                    }
-                                    _ => ()
-                                }
+                        if let TypeVariants::TyRef(_, ref ty_mut) = lhs_ty.sty {
+                            match self.lookup_op_method(expr, ty_mut.ty, vec![rhs_ty_var],
+                                        token::intern(name), trait_def_id,
+                                        lhs_expr) {
+                                Ok(_) => {
+                                    err.span_note(
+                                        lhs_expr.span,
+                                        &format!(
+                                            "this is a reference of type that `{}` can be \
```

Comment:
"reference of type that `+` can be applied to" sounds ungrammatical to me. I am not a native speaker, so I would like to invite others to chip in.